### PR TITLE
Renaming to deprecate LegacyEmailSubscriptions component.

### PR DESCRIPTION
**Pull request by weerd**
### What's this PR do?

This pull request renames the `EmailSubscriptions` component to `LegacyEmailSubcriptions` and will be replaced by a new component (the `NewsletterSubscriptions` component but that will be renamed to `EmailSubcriptions`). It also prefixes other associated resources and components related to this component that will also be replaced.

### How should this be reviewed?

👀 

### Any background context you want to provide?

Walking the walk after [RFC #16](https://github.com/DoSomething/rfcs/pull/16) proposed this legacy code workflow; so I'm trying it out!

### Relevant tickets

References [Pivotal #177781876](https://www.pivotaltracker.com/story/show/177781876).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

**Review by DFurnes — APPROVED** [review]
🚧

File: resources/assets/components/pages/AccountPage/Subscriptions/Subscriptions.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+
     <p className="text-gray-600">
       Our newsletters can help you stay informed, stay entertained, and even
       help you win scholarships.
     </p>
 
-    <EmailSubscriptions {...props} />
-    <div>
-      <CancelEmailSubscription {...props} />
-    </div>
```
[the remark this answers is not in the document]
Comment (weerd) [reply resources/assets/components/pages/AccountPage/Subscriptions/Subscriptions.js:18]: I double checked the layout and it did not perform any function 🤷‍♂️

[the remark this answers is not in the document]
Comment (weerd) [reply resources/assets/components/pages/AccountPage/Subscriptions/Subscriptions.js:18]: Oh, also meant to say @DFurnes that inside the `CancelEmailSubscription` it renders the `<p>` for the content inside an enclosing`<div>` already, so it was just redundant.

[the remark this answers is not in the document]
Comment (DFurnes) [reply resources/assets/components/pages/AccountPage/Subscriptions/Subscriptions.js:18]: Ah, div-ception! 🔲

**Pull request merged**